app/src/components/Home/components/WorkspaceCard/index.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState, useEffect } from "react";
+import { useSelector } from "react-redux";
+import { getUserAuthDetails } from "store/selectors";
+import { getAvailableTeams } from "store/features/teams/selectors";
+import { Col, Skeleton } from "antd";
+import { CreateWorkspaceView } from "./components/CreateWorkspaceView";
+import { TeamsListView } from "./components/TeamsListView";
+import { getPendingInvites } from "backend/workspace";
+import { isCompanyEmail } from "utils/FormattingHelper";
+import Logger from "lib/logger";
+import { Invite } from "types";
+
+export const TeamsCard: React.FC = () => {
+  const user = useSelector(getUserAuthDetails);
+  const availableTeams = useSelector(getAvailableTeams);
+  const [pendingInvites, setPendingInvites] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    if (!user.loggedIn) {
+      setIsLoading(false);
+      return;
+    }
+    getPendingInvites({ email: true, domain: true })
+      .then((res: any) => {
+        if (res?.pendingInvites) {
+          // Remove duplicate invites for the same team (A team can have multiple email invite and a domain invite)
+          const uniqueInvitesArray = res?.pendingInvites.reduce((acc: Invite[], invite: Invite) => {
+            const teamId = invite.metadata.teamId;
+            // Checking if an invite with the same teamId already exists in the accumulator
+            const existingInvite = acc.find((existing) => existing.metadata.teamId === teamId);
+            if (!existingInvite || invite.createdTs > existingInvite.createdTs) {
+              // If not found or the new invite is newer, replace or add the invite to the accumulator
+              const updatedAcc = existingInvite
+                ? acc.map((existing) => (existing.metadata.teamId === teamId ? invite : existing))
+                : [...acc, invite];
+
+              return updatedAcc;
+            }
+            return acc;
```

Comment:
We can make this more readable by rewriting it as
```ts
 if (res?.pendingInvites) {
  // Have a dictionary for invites where key is teamId and value is invite
  const uniqueInvites = {}

  // Sort the pending invites w.r.t to createdTs in ascending order and update the uniqueInvites dictionary
  // this will ensure that you are storing only the latest invite for a team
  res?.pendingInvites.sort((a,b) => ....).forEach(invite => uniqueInvites[invite.metadata.teamId] = invite)

  setPendingInvites(Object.values(uniqueInvites))
 }
```